### DISCUSSION TITLE
MacOS: Plugin No Longer Functions as of March 4th (or thereabouts)

### DISCUSSION THREAD

**Discussion by Tetraplasm** [fredemmott/StreamDeck-Discord#59] [Q&A] [answered]
**Describe the bug**
After (I believe) updating to Discord 0.0.261 recently, I (and some other users) are unable to use the stream deck plugin anymore. I made sure to get the most recent version of the plugin and confirm that my discord developer app was still in order, confirming that the client ID and secret were in order. Furthermore, I tried the "reconnect" and "reauthenticate" buttons on the stream deck configuration window, to no avail.

**To Reproduce**
Press the buttons on the stream deck, which produce a yellow ! triangle

**Expected behavior**
Buttons should function as expected (e.g. toggle self-mute)

**Debugging Information**

State: no info
Last state change: never

**Log File**
.discord0.log:
09:25:09.9210 Property inspector event: com.fredemmott.discord.rpc.getState
11:08:52.2256 Property inspector event: com.fredemmott.discord.rpc.getState
11:08:55.8684 Property inspector event: com.fredemmott.discord.rpc.getState
11:09:04.1964 Connecting to Discord
11:09:04.1966 Discord state change: rpcState = CONNECTING, muted = 1, deafened = 1
11:09:05.1244 Discord state change: rpcState = REQUESTING_USER_PERMISSION, muted = 1, deafened = 1
11:09:10.2686 Discord state change: rpcState = REQUESTING_ACCESS_TOKEN, muted = 1, deafened = 1
11:09:10.6502 Got new credentials for app id812762426487472191 - with oauth token = no
11:09:10.6590 Discord state change: rpcState = AUTHENTICATION_FAILED, muted = 1, deafened = 1
11:09:46.5054 Connecting to Discord
11:09:46.5056 Discord state change: rpcState = CONNECTING, muted = 0, deafened = 1
11:09:46.6237 Property inspector event: com.fredemmott.discord.rpc.reauthenticate
11:09:46.6237 Discord state change: rpcState = CONNECTION_FAILED, muted = 0, deafened = 1
11:09:46.6238 Will try to connect in 1 second...
11:09:46.6239 Connecting to Discord
11:09:46.6240 Discord state change: rpcState = CONNECTING, muted = 0, deafened = 0
11:09:47.0845 Discord state change: rpcState = REQUESTING_USER_PERMISSION, muted = 0, deafened = 0
11:09:49.1371 Discord state change: rpcState = REQUESTING_ACCESS_TOKEN, muted = 0, deafened = 0
11:09:49.3677 Got new credentials for app id812762426487472191 - with oauth token = yes
11:09:49.3869 Discord state change: rpcState = AUTHENTICATING_WITH_ACCESS_TOKEN, muted = 0, deafened = 0
11:09:51.3153 Discord state change: rpcState = REQUESTING_VOICE_STATE, muted = 0, deafened = 0
11:09:51.4197 Connected to Discord
11:09:51.6385 Discord state change: rpcState = READY, muted = 0, deafened = 0
11:09:51.6387 Discord state change: rpcState = READY, muted = 0, deafened = 0
11:09:54.6165 Property inspector event: com.fredemmott.discord.rpc.getState
11:09:56.9908 Property inspector event: com.fredemmott.discord.rpc.getState
11:09:57.7409 Property inspector event: com.fredemmott.discord.rpc.getState
11:10:04.3686 Key Up: com.fredemmott.discord.mute 3F67536B01B4E8906804DD15431326D9
11:10:04.4235 Discord state change: rpcState = READY, muted = 1, deafened = 0
11:10:05.0263 Key Up: com.fredemmott.discord.mute 3F67536B01B4E8906804DD15431326D9
11:10:05.0390 Discord state change: rpcState = READY, muted = 0, deafened = 0
11:10:05.8342 Key Up: com.fredemmott.discord.deafen 21795812F39B57F78760B5547BE28C45
11:10:05.8576 Discord state change: rpcState = READY, muted = 0, deafened = 1
11:10:06.5950 Key Up: com.fredemmott.discord.deafen 21795812F39B57F78760B5547BE28C45
11:10:06.6809 Discord state change: rpcState = READY, muted = 0, deafened = 0
11:10:44.1105 Property inspector event: com.fredemmott.discord.rpc.getState
11:10:45.1682 Property inspector event: com.fredemmott.discord.rpc.getState
15:44:06.5194 Discord state change: rpcState = READY, muted = 0, deafened = 0
09:30:34.5644 Property inspector event: com.fredemmott.discord.rpc.getState
09:30:37.7549 Property inspector event: com.fredemmott.discord.rpc.getState
09:30:39.9036 Property inspector event: com.fredemmott.discord.rpc.getState
09:30:50.9360 Property inspector event: com.fredemmott.discord.rpc.getState
09:30:51.3727 Property inspector event: com.fredemmott.discord.rpc.getState
09:30:53.0096 Property inspector event: com.fredemmott.discord.rpc.getState
10:06:30.6430 Property inspector event: com.fredemmott.discord.rpc.getState
10:06:31.7104 Property inspector event: com.fredemmott.discord.rpc.getState
10:41:44.1754 Property inspector event: com.fredemmott.discord.rpc.getState
10:50:12.7367 Property inspector event: com.fredemmott.discord.rpc.getState
10:50:13.9083 Property inspector event: com.fredemmott.discord.rpc.getState
14:50:58.4473 Property inspector event: com.fredemmott.discord.rpc.getState
18:21:58.0857 Key Up: com.fredemmott.discord.mute 0F0146366092C90C839FEAA0B4A77E47
18:21:58.3509 Discord state change: rpcState = READY, muted = 1, deafened = 0
18:22:01.9217 Key Up: com.fredemmott.discord.mute 0F0146366092C90C839FEAA0B4A77E47
18:22:02.0080 Discord state change: rpcState = READY, muted = 0, deafened = 0
18:48:47.8897 Key Up: com.fredemmott.discord.mute 0F0146366092C90C839FEAA0B4A77E47
18:48:48.0020 Discord state change: rpcState = READY, muted = 1, deafened = 0
18:55:27.7575 Key Up: com.fredemmott.discord.mute 0F0146366092C90C839FEAA0B4A77E47
18:55:27.9331 Discord state change: rpcState = READY, muted = 0, deafened = 0
18:56:03.1467 Key Up: com.fredemmott.discord.mute 4036A87BB33C7F9D896A0B57A62F7EBE
18:56:03.1970 Discord state change: rpcState = READY, muted = 1, deafened = 0
18:56:05.1401 Key Up: com.fredemmott.discord.mute 4036A87BB33C7F9D896A0B57A62F7EBE
18:56:05.2576 Discord state change: rpcState = READY, muted = 0, deafened = 0
19:17:06.5967 Key Up: com.fredemmott.discord.mute 0F0146366092C90C839FEAA0B4A77E47
19:17:06.7878 Discord state change: rpcState = READY, muted = 1, deafened = 0
19:17:10.0194 Key Up: com.fredemmott.discord.mute 0F0146366092C90C839FEAA0B4A77E47
19:17:10.1193 Discord state change: rpcState = READY, muted = 0, deafened = 0
19:32:51.9587 Key Up: com.fredemmott.discord.mute 0F0146366092C90C839FEAA0B4A77E47
19:32:52.0880 Discord state change: rpcState = READY, muted = 1, deafened = 0
19:32:55.7267 Key Up: com.fredemmott.discord.mute 0F0146366092C90C839FEAA0B4A77E47
19:32:55.8358 Discord state change: rpcState = READY, muted = 0, deafened = 0
19:32:57.6027 Key Up: com.fredemmott.discord.mute 0F0146366092C90C839FEAA0B4A77E47
19:32:57.7041 Discord state change: rpcState = READY, muted = 1, deafened = 0
19:33:11.4345 Key Up: com.fredemmott.discord.mute 0F0146366092C90C839FEAA0B4A77E47
19:33:11.4499 Discord state change: rpcState = READY, muted = 0, deafened = 0
22:00:07.4837 Key Up: com.fredemmott.discord.mute 0F0146366092C90C839FEAA0B4A77E47
22:00:07.6492 Discord state change: rpcState = READY, muted = 1, deafened = 0
22:00:15.2022 Key Up: com.fredemmott.discord.mute 0F0146366092C90C839FEAA0B4A77E47
22:00:15.4788 Discord state change: rpcState = READY, muted = 0, deafened = 0

.micmutetoggle0.log
17:09:42.2468 AudioMuteStreamDeckPlugin.cpp: Creating action com.fredemmott.micmutetoggle.toggle with context 7046B32808885A0DCB1517A0B42DD57F
15:01:57.5998 AudioMuteStreamDeckPlugin.cpp: Creating action com.fredemmott.micmutetoggle.mute with context 42838EF545B4745BBE068DEA8C018A03

** Operating System **
MacOS Catalina 10.15.7

**Additional context**
Stream Deck XL

**Comment by fredemmott** [comment]
Still works at least on Big Sur with latest Discord; have you followed the instructions in [TROUBLESHOOTING.md]?

> and some other users

Assumign you're referring to #57 that sounds unrelated: it mentions running as administrator, which suggests windows, and is known to cause problems.

**Comment by fredemmott** [comment, marked as answer]
If you're using a prerelease (v4.x), note #55

**Comment by Tetraplasm** [comment]
Yes! It turns out I had forgotten to simply restart both the stream deck plugin and discord. Though I must have at some point in the last few days, I didn't do so _after_ re-entering the credentials. My apologies for wasting your time. Thank you so much for all the work you do!

**Reply by fredemmott** [reply]
Thanks for the update - are you using the pre-release then? Restarting the stream deck plugin after entering credentials should not be necessary on 3.x
